client/app/cloud/project/compute/loadbalancer/configure/compute-loadbalancer-configure.controller.js

```diff
@@ -0,0 +1,269 @@
+            }
+        };
+
+        // Data
+        this.loadbalancer = {};
+        this.table = {
+            server: []
+        };
+
+        this.form = {
+            openstack: false,
+            servers: {}
+        };
+
+        this.toggle = {
+            updatedMessage: false
+        };
+
+    }
+
+    $onInit () {
+
+        // Get loadbalancer pending tasks and define poller
+        this.tasks = this.ControllerHelper.request.getArrayLoader({
+            loaderFunction: () => this.IpLoadBalancerTaskService.getTasks(this.loadbalancerId).then(tasks => _.filter(tasks, task => _.includes(["todo", "doing"], task.status))),
+            successHandler: () => this.startTaskPolling()
+        });
+        this.tasks.load();
+
+
+        let validatePromise;
+        // Terminate validation if params exists
+        if (this.validate) {
+            this.loaders.loadbalancer = true;
+            validatePromise = this.OvhApiCloudProjectIplb.Lexi().validate({ serviceName: this.serviceName, id: this.validate }, {}).$promise
+                .then(() => {
+                    this.$location.search("validate", null);
+                    this.toggle.updatedMessage = true;
+                })
+                .catch(err => this.CloudMessage.error([this.$translate.instant("cpc_loadbalancer_error"), err.data && err.data.message || ""].join(" ")))
+                .finally(() => { this.loaders.loadbalancer = false; });
+            this.validate = "";
+        } else {
+            validatePromise = Promise.resolve("");
+        }
+        // After validation, load the loadbalancer
+        validatePromise.then(() => this.getLoadbalancer(true));
+
+        this.$scope.$on("$destroy", () => this.stopTaskPolling());
+        this.initGuides();
+    }
+
+    initGuides () {
+        this.guides = {
+            horizon: this.ovhDocUrl.getDocUrl("g1773.creer_un_acces_a_horizon")
+        };
+    }
+
+    // Get cloud servers to add in the loadbalancer
+    getServers () {
+        if (this.loaders.table.server) {
+            return this.$q.reject("servers already loading");
+        }
+        this.loaders.table.server = true;
+        return this.$q.all({
+            cloudServers: this.OvhApiCloudProject.Instance().Lexi().query({ serviceName: this.serviceName }).$promise,
+            attachedServers: this.CloudProjectComputeLoadbalancerService.getAttachedServers(this.loadbalancer)
+        }).then(({ cloudServers, attachedServers }) => {
+            this.attachedServers = {};
+            _.forEach(attachedServers, attachedServer => {
+                if (attachedServer.status === "active") {
+                    this.attachedServers[attachedServer.address] = attachedServer;
+                }
+            });
+            this.form.servers = _.mapValues(this.attachedServers, e => !!e);
+            // Generate array of object type as {ipv4, name}
+            this.table.server = this.table.server.concat(
+                _.uniq(
```

Comment:
uniq uniot flatten map map filter.

That's easy to understand.

At this point if the data structure is so complicated you should comment & explain.  Annnddd I'm reapeating myself but all of this should not be in a controller